typing/parmatch.ml

```diff
@@ -1955,19 +1951,24 @@ and lubs ps qs = match ps,qs with
 (* Apply pressure to variants *)
 
 let pressure_variants tdefs patl =
+  ignore (pressure_variants
+            (Some tdefs)
+            (List.map (fun p -> [p; omega]) patl))
```

Comment:
We have a mismatch problem where the internal `pressure_variants` definition needs a fine-grained interface internally, but the external `pressure_variants` wants to expose a simpler interface and needs to bridge the gap.

In the previous version, we had an internal `pressure_variants` on just value patterns, and we exported a single `pressure_variants` that was in fact defined on computation patterns (it took care of splitting `exception` patterns away properly).

In the new version, I have two external variants: `pressure_variants` on value patterns (which didn't exist before), and then a `pressure_variants_in_computation_pattern` function on computation patterns (which is equivalent to the old external definition).

You would then expect that I now have *two* functions that have to do the ugly mapping from the external to the internal interface (wrapping `tdefs` in a `Some`, and doing Spaghetti-Monster-knows-what to those poor patterns). The trick I used is to define `pressure_variants_in_computation_pattern` in terms of the *external* function on value patterns, rather than in terms of the internal function. So now only the first external function has some ugly plumbing (it's the one you are commenting about). The second external function calls `split_pattern`, collects patterns in the expected way, and then calls the first external function on both sides.

My impression is that your analysis comes from looking at the diff in a wrong way (the change of names is confusing), rather than looking at the resulting code. I don't see the problem with this map, that does precisely the part of the previous function that was related to bridging the external-internal gap, separated from the `split_pattern`-related data churn.

Reply:
> We have a mismatch problem where the internal pressure_variants definition needs a fine-grained interface internally, but the external pressure_variants wants to expose a simpler interface and needs to bridge the gap.

I had indeed missed the fact that `pressure_variants` was used externally, I had assumed that only `pressure_variants_in_computation_pattern` would be.